tools/bcdb_mock/pkg/escrow/escrow.go

```diff
@@ -0,0 +1,332 @@
+	}
+	for _, escrowDetails := range rpi.Infos {
+		// in case of an error in this flow we continue, so we try to refund as much
+		// client as possible even if one fails
+		addressInfo, err := types.GetByAddress(ctx, e.db, escrowDetails.EscrowAddress)
+		if err != nil {
+			log.Error().Msgf("failed to load escrow address info: %s", err)
+			continue
+		}
+		kp, err := e.wallet.KeyPairFromSeed(addressInfo.Secret)
+		if err != nil {
+			log.Error().Msgf("failed to parse escrow address secret: %s", err)
+			continue
+		}
+		if err = e.wallet.Refund(*kp, id); err != nil {
+			log.Error().Msgf("failed to refund clients: %s", err)
+			continue
+		}
+	}
+	rpi.Canceled = true
+	if err = types.ReservationPaymentInfoUpdate(ctx, e.db, rpi); err != nil {
+		return errors.Wrapf(err, "could not mark escrows for %d as canceled", rpi.ReservationID)
+	}
+	return nil
+}
+
+// payoutFarmers pays out the farmer for a processed reservation
+func (e *Escrow) payoutFarmers(ctx context.Context, id schema.ID) error {
+	rpi, err := types.ReservationPaymentInfoGet(ctx, e.db, id)
+	if err != nil {
+		return errors.Wrap(err, "failed to get reservation escrow info")
+	}
+	// we already verified we have enough balance on every escrow for this reservation
+	for _, escrowDetails := range rpi.Infos {
+		// in case of an error in this flow we continue, so we try to pay as much
+		// farmers as possible even if one fails
+		farm, err := e.farmAPI.GetByID(ctx, e.db, int64(escrowDetails.FarmerID))
+		if err != nil {
+			log.Error().Msgf("failed to load farm info: %s", err)
+			continue
+		}
+		// TODO rework this type to an object with currency and filter based on "TFT"
+		destination := farm.WalletAddresses[0]
+		addressInfo, err := types.GetByAddress(ctx, e.db, escrowDetails.EscrowAddress)
+		if err != nil {
+			log.Error().Msgf("failed to load escrow address info: %s", err)
+			continue
+		}
+		kp, err := e.wallet.KeyPairFromSeed(addressInfo.Secret)
+		if err != nil {
+			log.Error().Msgf("failed to parse escrow address secret: %s", err)
+			continue
+		}
+		if err = e.wallet.PayoutFarmer(*kp, destination, escrowDetails.TotalAmount, id); err != nil {
+			log.Error().Msgf("failed to pay farmer: %s", err)
+			continue
+		}
+		// now refund any possible overpayment
+		if err = e.wallet.Refund(*kp, id); err != nil {
+			log.Error().Msgf("failed to pay farmer: %s", err)
+			continue
+		}
+	}
+	rpi.Released = true
+	if err = types.ReservationPaymentInfoUpdate(ctx, e.db, rpi); err != nil {
+		return errors.Wrapf(err, "could not mark escrows for %d as released", rpi.ReservationID)
+	}
+	return nil
+}
+
+// RegisterReservation registers a workload reservation
+func (e *Escrow) RegisterReservation(reservation workloads.Reservation) ([]types.EscrowDetail, error) {
+	job := reservationRegisterJob{
+		reservation:  reservation,
+		responseChan: make(chan reservationRegisterJobResponse),
+	}
+	e.reservationChannel <- job
+
+	response := <-job.responseChan
```

Comment:
If this is going to block anyway on the response, why is it needed to do it over channel. Can't it be something like 

```go
job.Process()
```